script.js

```diff
@@ -1,4 +1,120 @@
+//Global Variable
+// Initialize variables for player names, dice rolls, and orders
+var userName1 = "";
+var userName2 = "";
+var diceRoll1a = 0;
+var diceRoll1b = 0;
+var diceRoll2a = 0;
+var diceRoll2b = 0;
+var player1Order = "";
+var player2Order = "";
+
+// Function to generate a random dice roll
+function rollDice() {
+  return Math.floor(Math.random() * 6) + 1;
+}
+
+// Function to check if input is a valid order choice (1 or 2)
+function isValidOrderChoice(input) {
+  return input === "1" || input === "2";
+}
+
+// Function to reset the game
+function resetGame() {
+  diceRoll1a = 0;
+  diceRoll1b = 0;
+  diceRoll2a = 0;
+  diceRoll2b = 0;
+  player1Order = "";
+  player2Order = "";
+}
+
+// Function to play the game
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // Get player names
+  if (userName1 === "" && input !== "") {
+    userName1 = input;
+    return `Welcome Player 1: ${userName1}, please enter the username for Player 2.`;
+  }
+
+  if (userName2 === "" && input !== "") {
+    userName2 = input;
+    return `Welcome Player 2: ${userName2}, click submit to start the dice roll game.`;
+  }
+
+  // Check if both players have names and if dice rolls need to be performed
+  if (
+    userName1 !== "" &&
+    userName2 !== "" &&
+    diceRoll1a === 0 &&
+    diceRoll2a === 0
+  ) {
+    // Roll the dice for each player
+    diceRoll1a = rollDice();
+    diceRoll1b = rollDice();
+    diceRoll2a = rollDice();
+    diceRoll2b = rollDice();
+
+    // Prompt players to choose the order of their dice rolls
+    return `${userName1}, you rolled ${diceRoll1a} and ${diceRoll1b}.<br>${userName2}, you rolled ${diceRoll2a} and ${diceRoll2b}.<br>Now, both players choose the order for your dice rolls (1 or 2).`;
+  }
+
+  // Check if both players have chosen the order for their dice rolls
+  if (
+    userName1 !== "" &&
+    userName2 !== "" &&
+    diceRoll1a !== 0 &&
+    diceRoll2a !== 0 &&
+    player1Order === ""
+  ) {
+    // Player 1 chooses the order of dice rolls
+    if (isValidOrderChoice(input)) {
+      player1Order = input;
+      return `${userName2}, choose the order for your dice rolls (1 or 2).`;
+    }
+    return `${userName1}, please choose the order for your dice rolls (1 or 2).`;
```

Comment:
superb! this is wildddddly different from when you first started `if else`. Well done!